[SH-78] 방 생성 기능 추가

leewontae0 opened the pull request:
## 📌 관련 이슈
- Jira: [SH-78]

## 📝 PR 요약
방 생성 기능 추가

## ⚒️ 주요 변경 사항
<!-- 구체적인 변경 내용을 bullet point로 나열해주세요 -->
- "/rooms" 입력 시 web socket connection
- url parameter 를 통해 jwt 가져오기
- 방 생성 후 세션에 저장

## 🧪 테스트 체크리스트
- [ 방 생성 테스트 ] 

## ✅ 체크리스트
- [x] 로컬에서 테스트를 완료했나요?
- [x] Assignees를 등록했나요?
- [x] Label을 지정했나요?

[SH-78]: https://snowhite.atlassian.net/browse/SH-78?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

leewontae0 committed:
feat : 방 리스트 조회 기능 추가

- "room" get 요청 시 LobbyHandler::getRooms 호출
-

leewontae0 committed:
Merge branch 'SH-77' into feature

leewontae0 committed:
feat : 방 생성 기능 추가

- 방 생성 기능 추가
- wss://rooms 입력 시 websocket connection request
- url parameter 에서 jwt 파싱
- token decode를 통해 user identification

je0ng3 reviewed (commented):
컨트롤러 api 부분 타입 맞춰주시구
로컬테스트 에러가 발생해서 확인해주세요~
Parameter 0 of constructor in com.snowhite.server.service.LobbyService required a bean of type 'org.springframework.data.redis.core.ReactiveRedisTemplate' that could not be found.

leewontae0 committed:
fix : ReactiveRedisTemplate Bean 등록

leewontae0 committed:
fix : RedisConfig 수정

leewontae0 committed:
fix : 방 리스트 조회 기능 버그 수정

- ReactiveRedisTemplate 빈 등록 시 발생하던 에러 해결

leewontae0 committed:
Merge branch 'develop' into SH-77

leewontae0 committed:
style : LobbyService Qualifier annotation 수정

leewontae0 committed:
Merge branch 'SH-77' of
https://github.com/snowhite-hyu/backend into SH-77

leewontae0 committed:
fix : JwtWebFilter, SecurityConfig 컴파일 에러 수정

- JwtWebFilter에서 findById 함수의 인자로 String을 줌으로써
- 발생하는 에러 수정
- SecurityConfig , 대신 . 사용된 에러 수정

leewontae0 committed:
refactor : <String, Long> redis 삭제

- 기존에 roomId set을 별도로 redis에 저장했었으나,
- 패턴 매칭 기능을 통해 존재의의가 사라짐으로 인해 삭제

leewontae0 committed:
Merge branch 'SH-77' into SH-78

leewontae0 committed:
refactor : 방 생성 기능 리팩토링

- 추후에 다른 action을 처리할 때, 구조를 변경하지 않아도 되게 수정함
- user 당 websocket session을 가지고 있게 설계함
- redis에 websocketsession을 저장할 수 없어서, sessionId만 저장
- 메모리에 {sessionId:websocketsession} 저장
- web socket conncection 시에 저장하고,
- connection이 close될 때 삭제하도록 함

leewontae0 commented:
> 컨트롤러 api 부분 타입 맞춰주시구 로컬테스트 에러가 발생해서 확인해주세요~ Parameter 0 of constructor in com.snowhite.server.service.LobbyService required a bean of type 'org.springframework.data.redis.core.ReactiveRedisTemplate' that could not be found.

로컬테스트 에러 발생 안 하는 것 확인했습니다.
추가로 컨트롤러 도입에 관해서는 정확하진 않은데, STOMP 프로토콜 사용하지 않으면 Controller로 websocket handle 하는 것이 불가능해 보이긴 합니다.

> STOMP를 사용하면, 웹소켓만 사용할 때와 다르게 하나의 연결주소마다 핸들러 클래스를 따로 구현할 필요없이 Controller 방식으로 간편하게 사용할 수 있다.
출처: https://velog.io/@guswns3371/WebSocket-Spring

이것 말고도 여러모로 찾아봤는데, 대부분 STOMP에 사용하시더라구요.
어차피 대부분 web socket으로 통신하니까 그냥 router 형식으로 통일하는 게 나아보여요.

lheesoo1020 reviewed (approved):
일단 router, controller 통일은 뒤로 미뤄놔도 괜찮을 것 같아요 문제 없으면 머지하면 될듯합니다
게임 시작 구현하면서 stomp 사용 한번 고려해봐야겠네요

je0ng3 reviewed (commented):
실행되는건 확인했습니다! 그런데 스웨거에 url이 안뜨는 문제가 있는데 혹시 해결 가능하실까요? 일단 postman으로 테스트 가능한지 보려고  http://localhost:8080/lobby 주소로 보냈을 때는 500에러가 발생합니다 혹시 테스트 주소가 잘못되었을까요?

leewontae0 commented:
> 실행되는건 확인했습니다! 그런데 스웨거에 url이 안뜨는 문제가 있는데 혹시 해결 가능하실까요? 일단 postman으로 테스트 가능한지 보려고 http://localhost:8080/lobby 주소로 보냈을 때는 500에러가 발생합니다 혹시 테스트 주소가 잘못되었을까요?

swagger는 제가 사용을 안 해봐서 잘 모르겠습니다.
"/lobby" 에러 발생한 건 이슈 생성해 주시면 감사하겠습니다.

일단 확인해보니, room 객체 직렬화에서 ApiResponse 객체와 충돌이 발생했습니다.

je0ng3 commented:
> > 실행되는건 확인했습니다! 그런데 스웨거에 url이 안뜨는 문제가 있는데 혹시 해결 가능하실까요? 일단 postman으로 테스트 가능한지 보려고 http://localhost:8080/lobby 주소로 보냈을 때는 500에러가 발생합니다 혹시 테스트 주소가 잘못되었을까요?
> 
> swagger는 제가 사용을 안 해봐서 잘 모르겠습니다. "/lobby" 에러 발생한 건 이슈 생성해 주시면 감사하겠습니다.
> 
> 일단 확인해보니, room 객체 직렬화에서 ApiResponse 객체와 충돌이 발생했습니다.

제가 확인이 늦었네요ㅠㅠ 이미 pr올리신 것 같아서 이슈 따로 안올리겠습니다.
그럼 이 코드 실행에는 문제없으니까 승인하고 머지 하겠습니다!
